src/hooks/useApproveCallback.ts

```diff
@@ -75,13 +107,9 @@ export function useApproveCallback(
       console.error('no spender')
       return
     }
-
-    let useExact = false
-    const estimatedGas = await tokenContract.estimateGas.approve(spender, MaxUint256).catch(() => {
-      // general fallback for tokens who restrict approval amounts
-      useExact = true
-      return tokenContract.estimateGas.approve(spender, amountToApprove.quotient.toString())
-    })
+    if (!estimatedGas) {
+      return
+    }
 
     return tokenContract
       .approve(spender, useExact ? amountToApprove.quotient.toString() : MaxUint256, {
```

Comment:
see i thought we were using max approve by default, what was that other code you were showing us @moodysalem ?

Reply:
you're right, this code shows we do max approve

imo, this code has really surprising behavior though, the code computing the `amountToApprove` argument uses `Trade#maximumAmountIn` (just below this function). i would prefer if `maxApprove` was a boolean argument to the function that defaulted to false